spacy/matcher/matcher.pyx

```diff
@@ -69,7 +73,7 @@ cdef class Matcher:
         """
         return self._normalize_key(key) in self._patterns
 
-    def add(self, key, patterns, *_patterns, on_match=None):
+    def add(self, key, patterns, *, on_match=None, greediness: str=None):
```

Comment:
How would you feel about renaming this argument to just `greedy`? 

I feel like `greediness` could easily invite misspellings or confusion because of the `y` &rarr; `i` transformation (especially for non-native speakers). "greedy" as in "greedy matching" on the other hand is terminology people are familiar with.

If we're worried that `greedy` as a keyword argument sounds too much like a boolean flag, we could make it `greedy_{something}`? Although I don't think `greedy="FIRST"` is that bad.

Reply:
I don't mind too much either way. I initially did want to avoid just `greedy` because it indeed sounds like a boolean flag. But then on the other hand, the error message would be descriptive anough to help steer users in the correct direction:
> ValueError: [E947] Matcher.add received invalid 'greediness' argument: expected a value from ['FIRST', 'LONGEST'] but got: 'True'